src/main/java/org/testng/internal/ClassImpl.java

```diff
@@ -187,7 +188,10 @@ private Module newModule(Class<Module> module) {
             m_xmlTest, m_annotationFinder, m_objectFactory) };
       }
     } else {
-      result = new Object[] { getDefaultInstance() };
+      Object defaultInstance = getDefaultInstance();
+      if (defaultInstance != null) {
+        result = new Object[]{defaultInstance};
+      }
```

Comment:
I don't know. The objective was to skip the null value. I didn't check what is doing the code after but it returns an empty array what is expected.